mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -183,24 +191,18 @@ private[ml] trait DecisionTreeParams extends PredictorParams
  */
 private[ml] trait TreeClassifierParams extends Params {
 
+  // Impurity should be overriden when setting a default. This should be a def, but has
+  // to be a val to maintain the proper documentation.
   /**
-   * Criterion used for information gain calculation (case-insensitive).
-   * Supported: "entropy" and "gini".
-   * (default = gini)
    * @group param
    */
-  final val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
-    " information gain calculation (case-insensitive). Supported options:" +
-    s" ${TreeClassifierParams.supportedImpurities.mkString(", ")}",
-    (value: String) => TreeClassifierParams.supportedImpurities.contains(value.toLowerCase))
-
-  setDefault(impurity -> "gini")
+  val impurity: Param[String] = new Param(this, "", "")
```

Comment:
right, that's the point... we have loss-based default for gbts, in contrast to other trees